app/interactors/show_user.rb

```diff
@@ -0,0 +1,25 @@
+class ShowUser
+  include Interactor
+
+  def call
+    validate_input
+    execute
+    validate_output
+  end
+
+  private
+
+  def validate_input
+    context.fail!(error: "invalid user id") unless context.id
+  end
+
+  def execute
+    context.user = User.find(context.id)
+  rescue ActiveRecord::RecordNotFound
+    context.user = nil
+  end
+
+  def validate_output
+    context.fail!(error: "invalid user") unless context.user
```

Comment:
I think this is good as long as you can render the message on your controllers and the developer using your API knows what's going on. Another approach you can take is defining a base `Error` class that standardizes how you return errors through your interactors. :+1: